lib/Monad_WP/NonDetMonadVCG.thy

```diff
@@ -1214,6 +1214,15 @@ lemma hoare_vcg_conj_lift:
 lemmas hoare_vcg_conj_lift_pre_fix
   = hoare_vcg_conj_lift[where P=R and P'=R for R, simplified]
 
+lemma hoare_vcg_conj_lift':
+  assumes x: "\<lbrace>P\<rbrace> f \<lbrace>Q\<rbrace>"
+  assumes y: "\<lbrace>P'\<rbrace> f \<lbrace>Q'\<rbrace>"
+  shows      "\<lbrace>P and P'\<rbrace> f \<lbrace>\<lambda>rv s. Q rv s \<and> Q' rv s\<rbrace>"
```

Comment:
I don't think this lemma is used at the moment, and I think that that adding an `and` like this is fighting against the simplifier. Is anyone aware of a particular use case for it?

Reply:
I will remove this as part of my forthcoming PR for doReplyTransfer_invs’, which contains some other work cleaning up some of the lemmas in this PR

Reply:
Thanks for catching this. I was thinking about it and then forgot. I am more in favour of removing this, to be honest.